pkg/core/mempool/subscriptions.go

```diff
@@ -0,0 +1,70 @@
+package mempool
+
+import (
+	"github.com/nspcc-dev/neo-go/pkg/core/transaction"
+)
+
+type EventType byte
+
+const (
+	TransactionAdded   EventType = 0x01
+	TransactionRemoved EventType = 0x02
+)
+
+type MemPoolEvent struct {
+	Type EventType
+	Tx   *transaction.Transaction
+	Data interface{}
+}
+
+func (mp *Pool) RunSubscriptions() {
+	if mp.subscriptionsEnabled.Load() {
+		go mp.notificationDispatcher()
+	}
+}
+
+func (mp *Pool) StopSubscriptions() {
+	if mp.subscriptionsEnabled.Load() {
+		close(mp.stopCh)
+	}
+}
+
+func (mp *Pool) SubscribeForTransactions(ch chan<- MemPoolEvent) {
+	if mp.subscriptionsEnabled.Load() {
+		mp.subCh <- ch
+	}
+}
+
+func (mp *Pool) UnsubscribeFromTransactions(ch chan<- MemPoolEvent) {
+	if mp.subscriptionsEnabled.Load() {
+		mp.unsubCh <- ch
+	}
+}
+
+// notificationDispatcher manages subscription to events and broadcasts new events.
+func (mp *Pool) notificationDispatcher() {
+	var (
+		// These are just sets of subscribers, though modelled as maps
+		// for ease of management (not a lot of subscriptions is really
+		// expected, but maps are convenient for adding/deleting elements).
+		txFeed = make(map[chan<- MemPoolEvent]bool)
+	)
+	for {
+		select {
+		case <-mp.stopCh:
+			return
+		case sub := <-mp.subCh:
+			txFeed[sub] = true
+		case unsub := <-mp.unsubCh:
+			delete(txFeed, unsub)
+		case event := <-mp.events:
+			// We don't want to waste time looping through transactions when there are no
+			// subscribers.
+			if len(txFeed) != 0 {
```

Comment:
You have a bit different situation here (per-transaction events) and this `if` doesn't really help you, so it can be removed.